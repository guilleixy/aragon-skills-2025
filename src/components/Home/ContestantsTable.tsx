"use client";

import { useState, useEffect } from "react";
import { Button, TextInput, Loader, Table } from "@mantine/core";

export default function ContestantsTable() {
  const [name, setName] = useState("");
  const [year, setYear] = useState("");
  const [escuela, setEscuela] = useState("");
  const [especialidad, setEspecialidad] = useState("");
  const [posicion, setPosicion] = useState("");
  const [loading, setLoading] = useState(false);
  const [results, setResults] = useState<any[]>([]);

  const fetchResults = async () => {
    setLoading(true);

    const params = new URLSearchParams();

    if (name) params.append("name", name.toUpperCase());
    if (year) params.append("year", year.toUpperCase());
    if (escuela) params.append("escuela", escuela.toUpperCase());
    if (especialidad) params.append("especialidad", especialidad.toUpperCase());
    if (posicion) params.append("posicion", posicion.toUpperCase());

    const res = await fetch(`/api/results?${params}`);
    const data = await res.json();

    setResults(data);
    setLoading(false);
  };

  useEffect(() => {
    fetchResults();
  }, []);

  return (
    <section
      id="table"
      style={{
        display: "block",
        height: "100%",
        marginTop: "150vh",
      }}
    >
      <h1>Ganadores</h1>
      <div
        style={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          flexWrap: "wrap",
          gap: "5rem",
        }}
      >
        <TextInput
          label="Nombre"
          value={name}
          onChange={(e) => setName(e.currentTarget.value)}
        />
        <TextInput
          label="A침o"
          value={year}
          onChange={(e) => setYear(e.currentTarget.value)}
          color="danger"
        />
        <TextInput
          label="Escuela"
          value={escuela}
          onChange={(e) => setEscuela(e.currentTarget.value)}
        />
        <TextInput
          label="Especialidad"
          value={especialidad}
          onChange={(e) => setEspecialidad(e.currentTarget.value)}
        />
        <TextInput
          label="Posici칩n"
          value={posicion}
          onChange={(e) => setPosicion(e.currentTarget.value)}
        />
        <Button
          style={{ marginTop: "25px" }}
          onClick={fetchResults}
          color="black"
        >
          Buscar
        </Button>
      </div>

      {loading ? (
        <Loader mt="lg" />
      ) : results.length > 0 ? (
        <Table mt="lg" highlightOnHover striped>
          <thead>
            <tr>
              <th>Nombre</th>
              <th>Escuela</th>
              <th>Especialidad</th>
              <th>A침o</th>
              <th>Posici칩n</th>
            </tr>
          </thead>
          <tbody>
            {results.map((res, key) => (
              <tr
                key={key}
                style={{ display: "flex", gap: "2rem", flexWrap: "wrap" }}
              >
                <td>{res.competidor.name}</td>
                <td>{res.competidor.escuela.name}</td>
                <td>{res.competidor.especialidad.name}</td>
                <td>{res.edicion.year}</td>
                <td>{res.position}</td>
                {/*                 <td>
                  <Button color="black" onClick={() => console.log("a")}>
                    Borrar
                  </Button>
                </td> */}
              </tr>
            ))}
          </tbody>
        </Table>
      ) : (
        <p style={{ marginTop: "30px" }}>No se encontraron resultados</p>
      )}
    </section>
  );
}
