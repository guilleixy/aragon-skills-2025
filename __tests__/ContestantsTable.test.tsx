import renderer from "react-test-renderer";
import { useState, useEffect } from "react";
import { Button, TextInput, Loader, Table } from "@mantine/core";
import ContestantsTable from "@/components/Home/ContestantsTable";

jest.mock("@mantine/core");

const renderTree = (tree: any) => renderer.create(tree);
describe("<ContestantsTable>", () => {
  it("should render component", () => {
    expect(renderTree(<ContestantsTable />).toJSON()).toMatchSnapshot();
  });
});
