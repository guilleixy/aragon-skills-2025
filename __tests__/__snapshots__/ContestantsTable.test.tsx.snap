// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContestantsTable> should render component 1`] = `null`;
